server/src/main/java/org/apache/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -499,6 +503,23 @@ public boolean doMonitor(ServiceEmitter emitter)
       final ServiceMetricEvent.Builder builder = new ServiceMetricEvent.Builder();
       MonitorUtils.addDimensionsToBuilder(builder, dimensions);
       emitter.emit(builder.build("jetty/numOpenConnections", ACTIVE_CONNECTIONS.get()));
+      ThreadPool jettyServerThreadPool = JETTY_SERVER_THREAD_POOL.get();
+      emitter.emit(builder.build("jetty/threadPoolNumTotalThreads", jettyServerThreadPool.getThreads()));
+      emitter.emit(builder.build("jetty/threadPoolNumIdleThreads", jettyServerThreadPool.getIdleThreads()));
+      emitter.emit(builder.build("jetty/threadPoolIsLowOnThreads", jettyServerThreadPool.isLowOnThreads() ? 1 : 0));
+      if (jettyServerThreadPool instanceof ThreadPool.SizedThreadPool) {
```

Comment:
when will this be true? As in, where is threadpool of this type being configured?

Reply:
https://github.com/apache/druid/blob/master/server/src/main/java/org/apache/druid/server/initialization/jetty/JettyServerModule.java#L210

Hi Abhishek, QueuedThreadPool implements SizedThreadPool which in turn implements ThreadPool, here I am using the minimal casting to use methods I need in case in the future the implementation of the thread pool changes then at least some of the metrics still work, I guess I can also just check if jettyServerThreadPool instanceof QueuedThreadPool because it is the only implementation now.

Reply:
> I guess I can also just check if jettyServerThreadPool instanceof QueuedThreadPool because it is the only implementation now.

that makes sense. I am thinking that we would want to report the queue size and busy threads instead of min and max threads.

Reply:
Thanks! 1) The min and max threads are just for operation convenience, their value don't change but it's very useful to plot them in the same graph as busy busy threads, for example, to get an idea on the usage, after all, it's not convenient to check druid runtime.properties each time to get the max and min value. Uses don't need them can just not ignore these two metrics. 2) I have simplified the logic to just use QueuedThreadPool to get rid of all the unnecessary checks.